### PR TITLE
Abstract the spi of shardingsphere driver url

### PR DESCRIPTION
**Pull request by totalo**
For #23388.

Changes proposed in this pull request:
  - Abstract the spi of shardingsphere driver url

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     */
+    public static byte[] getContent(final String url) {
+        for (ShardingsphereDriverURLProvider each : ShardingSphereServiceLoader.getServiceInstances(ShardingsphereDriverURLProvider.class)) {
+            if (each.accept(url)) {
+                return each.getContent(url);
+            }
+        }
+        throw new IllegalArgumentException("No suitable driver url provider for " + url);
+    }
+
```
Comment (strongduanmu) [inline jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java:41]: Please remove useless black line.

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+     * @param url the driver url
+     * @return the config content
+     */
+    public static byte[] getContent(final String url) {
+        for (ShardingsphereDriverURLProvider each : ShardingSphereServiceLoader.getServiceInstances(ShardingsphereDriverURLProvider.class)) {
+            if (each.accept(url)) {
+                return each.getContent(url);
+            }
+        }
+        throw new IllegalArgumentException("No suitable driver url provider for " + url);
```
Comment (strongduanmu) [inline jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java:39]: Can we add a new Exception with error code? Just like ColumnLabelNotFoundException.

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/spi/AbsolutePathDriverURLProvider.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public byte[] getContent(final String url) {
+        String configuredFile = url.substring("jdbc:shardingsphere:".length(), url.contains("?") ? url.indexOf("?") : url.length());
+        String file = configuredFile.substring(PATH_TYPE.length());
+        Preconditions.checkArgument(!file.isEmpty(), "Configuration file is required in ShardingSphere driver URL.");
+        try (InputStream stream = Files.newInputStream(new File(file).toPath())) {
+            Objects.requireNonNull(stream, String.format("Can not find configuration file `%s`.", file));
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8));
+            StringBuilder builder = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
```
Comment (strongduanmu) [inline jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/spi/AbsolutePathDriverURLProvider.java:57]: Please put null on the left condition.

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingsphereDriverURLProvider.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.driver.jdbc.core.driver;
+
+/**
+ * Shardingsphere driver URL provider.
+ */
+public interface ShardingsphereDriverURLProvider {
```
Comment (strongduanmu) [inline jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingsphereDriverURLProvider.java:23]: Can we add this new SPI interface to dev manual document?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java (hunk 1 above)
Comment (totalo) [reply jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java:41]: Ok

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java (hunk 2 above)
Comment (totalo) [reply jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURLManager.java:39]: done.

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/spi/AbsolutePathDriverURLProvider.java (hunk 3 above)
Comment (totalo) [reply jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/spi/AbsolutePathDriverURLProvider.java:57]: done.

File: jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingsphereDriverURLProvider.java (hunk 4 above)
Comment (totalo) [reply jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingsphereDriverURLProvider.java:23]: Yes, i'll add this in next pr.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**